core/src/main/java/org/apache/accumulo/core/metadata/StoredTabletFile.java

```diff
@@ -291,9 +291,28 @@ public TabletFileCq(Path path, Range range) {
     }
   }
 
-  private static class TabletFileCqMetadataGson {
+  static class TabletFileCqMetadataGson {
     private String path;
     private byte[] startRow;
     private byte[] endRow;
+
+    TabletFileCqMetadataGson() {}
+
+    TabletFileCqMetadataGson(AbstractTabletFile<?> atf) {
+      path = Objects.requireNonNull(atf.path.toString());
+      startRow = encodeRow(atf.range.getStartKey());
+      endRow = encodeRow(atf.range.getEndKey());
+    }
+
+    ReferencedTabletFile toReferencedTabletFile() {
+      return new ReferencedTabletFile(new Path(URI.create(path)),
+          new Range(decodeRow(startRow), true, decodeRow(endRow), false));
+    }
+
+    StoredTabletFile toStoredTabletFile() {
+      return new StoredTabletFile(new TabletFileCq(new Path(URI.create(path)),
```

Comment:
You could use this [static](https://github.com/apache/accumulo/blob/05fe0713f778e15be08c1ae16ddd8136970bc446/core/src/main/java/org/apache/accumulo/core/metadata/StoredTabletFile.java#L175-L177) helper method to create the StoredTabletFile which would make it a little simpler.